recipe/meta.yaml

```diff
@@ -43,8 +42,8 @@ requirements:
     - pip
```

Comment:
We could update to cython 3 based on https://github.com/apache/arrow/blob/apache-arrow-14.0.2/python/setup.py#L43-L45.